torch/lib/TensorLib/CPUGenerator.cpp

```diff
@@ -0,0 +1,36 @@
+#include "TensorLib/CPUGenerator.h"
+
+#define const_generator_cast(generator) \
+  dynamic_cast<const CPUGenerator&>(generator)
+
+namespace tlib {
+
+CPUGenerator::CPUGenerator(Context * context_)
```

Comment:
Could we not call this CPUGenerator? It is very confusing and up until I had gotten to this patch I thought it had to do with generating new instances of tensors. Maybe CpuRng?